### PR TITLE
fix: Remove console warnings in cache-related code

### PR DESCRIPTION
**Pull request by rafpaf**
This PR removes:
- a warning about missing React keys in the `Schedule` component
- a warning about invalid props in the `StrategyFormLauncher` component

### CODE REVIEW

**Comment by rafpaf** [comment]
* **#44173** <a href="https://app.graphite.dev/github/pr/metabase/metabase/44173?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: frontend/src/metabase/common/utils/doNotForwardProps.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const doNotForwardProps = (...propNamesToBlock: string[]) => ({
+  shouldForwardProp: (propName: string) => !propNamesToBlock.includes(propName),
```
Comment (sloansparger) [inline frontend/src/metabase/common/utils/doNotForwardProps.ts:2]: !nit use `Set` any time you can 🙏

**Review by sloansparger — APPROVED** [review]
approved, but if you want your code to be blazingly fast ™️ you should use a Set 😈

File: frontend/src/metabase/common/utils/doNotForwardProps.ts (hunk 1 above)
Comment (rafpaf) [reply frontend/src/metabase/common/utils/doNotForwardProps.ts:2]: I think that's a good policy and I'm kicking myself a tiny bit for not thinking of that. Perhaps it's an over-optimization though given the typical length of the array (fewer than 5 props).

**Pull request merged**